servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ServerDuplexHandler.java

```diff
@@ -143,7 +143,7 @@ private NettyH2HeadersToHttpHeaders h2HeadersToH1HeadersServer(Http2Headers h2He
                         h2Headers.add(TRANSFER_ENCODING, CHUNKED);
                     }
                 }
-            } else if (contentLength >= 0L) {
+            } else if (contentLength != null) {
```

Comment:
since we don't need the value lets use `contains` to avoid serialization.

```diff
diff --git a/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java b/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java
index 61b22e7f3..7f90cfd10 100644
--- a/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java
+++ b/servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1ClientDuplexHandler.java
@@ -166,8 +166,8 @@ final class H2ToStH1ClientDuplexHandler extends AbstractH2DuplexHandler {
         h2HeadersSanitizeForH1(h2Headers);
         if (httpStatus != null) {
             final int statusCode = httpStatus.code();
-            final Long contentLength = h2Headers.getLong(CONTENT_LENGTH);
-            if (contentLength == null) {
+            final boolean containsContentLength = h2Headers.contains(CONTENT_LENGTH);
+            if (!containsContentLength) {
                 if (fullResponse) {
                     if (responseMayHaveContent(statusCode, method)) {
                         h2Headers.set(CONTENT_LENGTH, ZERO);
@@ -176,9 +176,8 @@ final class H2ToStH1ClientDuplexHandler extends AbstractH2DuplexHandler {
                     h2Headers.add(TRANSFER_ENCODING, CHUNKED);
                 }
             } else if (!responseMayHaveContent(statusCode, method)) {
-                throw new IllegalArgumentException("content-length (" + contentLength +
-                        ") header is not expected for status code " + statusCode + " in response to " + method.name() +
-                        " request");
+                throw new IllegalArgumentException("content-length header is not expected for status code " +
+                        statusCode + " in response to " + method.name() + " request");
             }
         }
         return new NettyH2HeadersToHttpHeaders(h2Headers, headersFactory.validateCookies());

```

Reply:
also same issue in `H2ToStH1ServerDuplexHandler` IIUC?